### ISSUE TITLE
False negative "Type alias "Point" cannot be used in this context"

### ISSUE THREAD

**Issue by rbusquet** [microsoft/pyright#6169]
**Describe the bug**
I believe there's a bug introduced on #5663 that's causing errors on valid code. The following snippet works correctly at runtime on python 3.12.

**Code or Screenshots**

```python
type Point = tuple[int, int]


def function() -> dict[Point, int]:
    heights = dict[Point, int]()
    return heights
```

![image](https://github.com/microsoft/pyright/assets/7198302/db0a59e1-c273-4e48-ab36-f9645286b23b)
.

**VS Code extension or command-line**
Using VS Code extension

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

This check was intended to catch the case where the type alias is used directly for a constructor call (e.g. `Point()`). It shouldn't apply when the type alias is used as a type argument in the LHS of the call expression.

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
